### PR TITLE
HTML Injection KB entry

### PR DESCRIPTION
**Pull request by BlueSquare1**

### CODE REVIEW

File: WEB_SERVICE/WEB/_MEDIUM/HTML_INJECTION/description.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
+### HTML Injection
+
+HTML injection is a security vulnerability that arises when an input point in a web application can be manipulated by a user, enabling them to insert arbitrary HTML code into a susceptible web page. Exploiting this vulnerability can lead to various attacks, such as leaking a user's session cookies, which can then be exploited for more attacks. Furthermore, it allows the attacker to alter the content displayed to the victims, granting them the ability to modify the page to insert malicious code or deface the page with their own message.
```
Comment (3asm) [inline WEB_SERVICE/WEB/_MEDIUM/HTML_INJECTION/description.md:3]: Didn't we want this for Mobile Application, the context should apply to Mobile App, hence the code below `Dart` `Kotlin` ...

File: WEB_SERVICE/WEB/_MEDIUM/HTML_INJECTION/description.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+        webView = findViewById(R.id.webView)
+
+        val name = intent.getStringExtra("name")
+
+        val html = "<html><body><h1>Hello, $name!</h1></body></html>"
+        webView.loadDataWithBaseURL(null, html, "text/html", "UTF-8", null)
+
+        val webSettings: WebSettings = webView.settings
+        webSettings.javaScriptEnabled = true
```
Comment (3asm) [inline WEB_SERVICE/WEB/_MEDIUM/HTML_INJECTION/description.md:78]: Should be set to False to denote it is not an XSS.

File: WEB_SERVICE/WEB/_MEDIUM/HTML_INJECTION/description.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        webView = WKWebView(frame: view.bounds)
+        webView.navigationDelegate = self
+        view.addSubview(webView)
+        
+        let name = "John Doe"
+        let html = "<html><body><h1>Hello, \(name)!</h1></body></html>"
+        webView.loadHTMLString(html, baseURL: nil)
```
Comment (3asm) [inline WEB_SERVICE/WEB/_MEDIUM/HTML_INJECTION/description.md:101]: Should disable XSS to denote it is not an XSS.

File: WEB_SERVICE/WEB/_MEDIUM/HTML_INJECTION/meta.json [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+      "HTML Injection (Imperva)":"https://www.imperva.com/learn/application-security/html-injection",
+   },
+   "title":"HTML Injection Vulnerability",
+   "privacy_issue":false,
+   "security_issue":true,
+   "categories":{
+      "OWASP_TOP_10":[
+         "A3:2017-Sensitive Data Exposure",
+         "A7:2017-Cross-Site Scripting (XSS)"
+      ],
```
Comment (3asm) [inline WEB_SERVICE/WEB/_MEDIUM/HTML_INJECTION/meta.json:15]: We don't have these.

File: WEB_SERVICE/WEB/_MEDIUM/HTML_INJECTION/meta.json [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+      "OWASP_MASVS_L1":[
+         "MSTG_PLATFORM_5",
+         "MSTG_PLATFORM_7"
+      ],
+      "OWASP_MASVS_L2":[
+         "MSTG_PLATFORM_5",
+         "MSTG_PLATFORM_7"
```
Comment (3asm) [inline WEB_SERVICE/WEB/_MEDIUM/HTML_INJECTION/meta.json:22]: Are you sure 5 and 7 are in the L1 and L2?

File: WEB_SERVICE/WEB/_MEDIUM/HTML_INJECTION/recommendation.md [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+
+- Use Templating Engines: Utilize secure templating engines or frameworks that automatically escape user input by default, such as Handlebars, Twig, or Django templates. These engines help prevent HTML injection by properly handling user input.
+
+- Whitelisting Input: Implement whitelisting or allow listing approaches to validate and restrict input for HTML elements, attributes, and protocols. Only allow specific tags, attributes, and protocols that are necessary for the application's functionality.
+
+- Contextual Output Encoding: Encode user-generated content based on its context. Different contexts, such as HTML attributes, JavaScript code, or CSS styles, require specific encoding techniques to prevent HTML injection attacks. Use appropriate encoding functions or libraries based on the context.
+
+- Content Security Policy (CSP): Implement and enforce a Content Security Policy for your web application. CSP helps mitigate HTML injection by specifying which sources of content (e.g., scripts, styles, images) are allowed to be loaded and executed on a webpage.
```
Comment (3asm) [inline WEB_SERVICE/WEB/_MEDIUM/HTML_INJECTION/recommendation.md:12]: CSP do not prevent HTML injection!!!

**Review by 3asm — CHANGES_REQUESTED** [review]

File: WEB_SERVICE/WEB/_MEDIUM/HTML_INJECTION/meta.json (hunk 5 above)
Comment (BlueSquare1) [reply WEB_SERVICE/WEB/_MEDIUM/HTML_INJECTION/meta.json:22]: They are, according to this:
https://mobile-security.gitbook.io/masvs/security-requirements/0x11-v6-interaction_with_the_environment

File: MOBILE_CLIENT/COMMON/_MEDIUM/HTML_INJECTION/recommendation.md [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+        Uri.dataFromString(sanitizedHtml, mimeType: 'text/html', encoding: Encoding.getByName('utf-8'))!.toString(),
+      );
+    }
+  }
+
+  String _sanitizeHtml(String html) {
+    // Perform necessary sanitization on the HTML string
+    // Here's a basic example using the html package:
+    final sanitized = html.replaceAll('<', '&lt;').replaceAll('>', '&gt;');
+    return sanitized;
```
Comment (3asm) [inline MOBILE_CLIENT/COMMON/_MEDIUM/HTML_INJECTION/recommendation.md:71]: This is not sufficient, please provide a complete fix, typically using a library that does HTML escape.

**Review by 3asm — CHANGES_REQUESTED** [review]

**Review by 3asm — APPROVED** [review]

**Pull request merged**